lib/internal/assert/calltracker.js

```diff
@@ -21,66 +23,110 @@ const {
 
 const noop = FunctionPrototype;
 
+class CallTrackerContext {
+  #expected;
+  #calls;
+  #name;
+  #stackTrace;
+  constructor({ expected, stackTrace, name }) {
+    this.#calls = [];
+    this.#expected = expected;
+    this.#stackTrace = stackTrace;
+    this.#name = name;
+  }
+
+  track(thisArg, args) {
+    ArrayPrototypePush(this.#calls, { thisArg, arguments: args });
+  }
+
+  get delta() {
+    return this.#calls.length - this.#expected;
+  }
+
+  reset() {
+    this.#calls = [];
+  }
+  getCalls() {
+    return this.#calls;
```

Comment:
as-is, this exposes a private mutable array. this will fix it:
```suggestion
    return ArrayPrototypeSlice(this.#calls);
```